realscale.py

```diff
@@ -65,16 +98,25 @@ def effect(self):
         p2_y = path[0][1][1][1]
 
         p_length = self.getUnittouu(str(distance((p1_x, p1_y),(p2_x, p2_y))) + self.getDocumentUnit())
+        
+        # calculate scaling center
+        center = self.calc_scale_center(p1_x, p1_y, p2_x, p2_y)
 
         #calculate scaling factor
         target_length = self.getUnittouu(str(self.options.length) + self.options.unit)
         factor = target_length / p_length
 
         # Get drawing and current transformations
         for obj in (scalepath, drawing):
-            # Scale both objects as desired
+            # Scale both objects about the center# first translate back to origin
+            scale_matrix = [[1, 0.0, -center[0]], [0.0, 1, -center[1]]]
+            simpletransform.applyTransformToNode(scale_matrix, obj)
+            # Then scale
             scale_matrix = [[factor, 0.0, 0.0], [0.0, factor, 0.0]]
-            simpletransform.applyTransformToNode(scale_matrix,obj)
+            simpletransform.applyTransformToNode(scale_matrix, obj)
+            # Then translate back to original scale center location
+            scale_matrix = [[1, 0.0, center[0]], [0.0, 1, center[1]]]
+            simpletransform.applyTransformToNode(scale_matrix, obj)
 
```

Comment:
It's a lot easier to do this in three steps - good thought :) (I had been thinking about how the ... I need to build the matrix...)